transaction/transaction.go

```diff
@@ -778,6 +757,42 @@ func AssignGroupID(txns []types.Transaction, account string) (result []types.Tra
 	return result, nil
 }
 
+// AddLease adds the passed lease (see types/transaction.go) to the header of the passed transaction
+// - txn: the types.Transaction to modify
+// - lease: the [32]byte lease to add to the header
+// - feePerByte: the new feePerByte
+func AddLease(tx types.Transaction, lease [32]byte, feePerByte uint64) (types.Transaction, error) {
```

Comment:
Consider making these two a method of `Transaction`

Reply:
@rotemh Just tried implementing that, and I liked how it looked, but it causes an import cycle that I don't see a way around. Doing that either imports `crypto` or `transaction` into the `types` package, but `crypto` and `transaction` import `types.`